Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -194,7 +194,7 @@ namespace detail {
 template <typename intersection_t, typename logger_t = std::false_type>
 bool checkIntersection(const intersection_t& intersection, double pLimit,
                        double oLimit, double tolerance,
-                       logger_t logger = logger_t{}) {
+                       [[maybe_unused]] logger_t logger = logger_t{}) {
```

Comment:
Unfortunately I don't think this works everywhere yet.

In other spaces we had to use 
```cpp
(void)logger;
```
in the function body to suppress the error.

Reply:
Hmm but it seems to build with the `[[maybe_unused]]` attribute...

Reply:
Ok. If the CI is happy, then let's not bother.